src/k8s/pkg/component/component_test.go

```diff
@@ -0,0 +1,243 @@
+package component
+
+import (
+	"flag"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/canonical/k8s/pkg/snap/mock"
+	"github.com/stretchr/testify/assert"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/registry"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"helm.sh/helm/v3/pkg/time"
+)
+
+var verbose = flag.Bool("test.log", false, "enable test logging")
+
+func actionConfigFixture(t *testing.T) *action.Configuration {
+	t.Helper()
+
+	registryClient, err := registry.NewClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &action.Configuration{
+		Releases:       storage.Init(driver.NewMemory()),
+		KubeClient:     &kubefake.FailingKubeClient{PrintingKubeClient: kubefake.PrintingKubeClient{Out: io.Discard}},
+		Capabilities:   chartutil.DefaultCapabilities,
+		RegistryClient: registryClient,
+		Log: func(format string, v ...interface{}) {
+			t.Helper()
+			if *verbose {
+				t.Logf(format, v...)
+			}
+		},
+	}
+}
+
+type chartOptions struct {
+	*chart.Chart
+}
+
+type chartOption func(*chartOptions)
+
+func buildChart(opts ...chartOption) *chart.Chart {
+	c := &chartOptions{
+		Chart: &chart.Chart{
+			Metadata: &chart.Metadata{
+				APIVersion: "v1",
+				Name:       "hello",
+				Version:    "0.1.0",
+			},
+		},
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c.Chart
+}
+
+func namedReleaseStub(name string, status release.Status) *release.Release {
+	now := time.Now()
+	return &release.Release{
+		Name: name,
+		Info: &release.Info{
+			FirstDeployed: now,
+			LastDeployed:  now,
+			Status:        status,
+			Description:   "Named Release Stub",
+		},
+		Config:  map[string]interface{}{"name": "value"},
+		Version: 1,
+	}
+}
+
+// "Mock" Initializer
+type MockHelmClientInitializer struct {
+	actionConfig *action.Configuration
+}
+
+func (r *MockHelmClientInitializer) InitializeHelmClientConfig() (*action.Configuration, error) {
+	return r.actionConfig, nil
+}
+
+func makeMeSomeReleases(store *storage.Storage, t *testing.T) {
+	t.Helper()
+	relStub1 := namedReleaseStub("whiskas-1", release.StatusDeployed)
+	relStub2 := namedReleaseStub("whiskas-2", release.StatusDeployed)
+	relStub3 := namedReleaseStub("whiskas-3", release.StatusDeployed)
+
+	for _, rel := range []*release.Release{relStub1, relStub2, relStub3} {
+		if err := store.Create(rel); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	all, err := store.ListReleases()
+	assert.NoError(t, err)
+	assert.Len(t, all, 3, "sanity test: three items added")
```

Comment:
I understand this is using testify, but let's keep `gomega` for all our tests. Have a look at https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/database/kubernetes_auth_tokens_test.go in order to adjust this

```suggestion
	assert.NoError(t, err)
	assert.Len(t, all, 3, "sanity test: three items added")
```

Reply:
Thanks for the example!